app.py

```diff
@@ -1,24 +1,47 @@
 import os
-from urllib.request import urlopen
-# from urllib.error import URLError, HTTPError
+from urllib.request import Request, urlopen
+from urllib.error import URLError, HTTPError
 from bs4 import BeautifulSoup
 from dotenv import load_dotenv
-# import sys
 load_dotenv()
 
 
-# TODO Send Push Notification
+# TODO send push notification
 
 KINDLE = os.getenv('KINDLE')
 
 
+def isURL(url):
+    isURL = url.find("https://www.amazon.com")
+    invalid = -1
+
+    if isURL != invalid:
+        return True
+    return False
+
+
 def url_to_html(url):
-    response = urlopen(url)
-    return response.read()
+    isAmazonURL = isURL(url)
+
+    if isAmazonURL:
+        try:
+            req = Request(url)
+            response = urlopen(req)
+            return response.read()
+        except HTTPError as e:
+            print(e.status, e.reason)
+        except URLError as e:
+            print(e.reason)
+
+    return False
 
 
-def html_to_soup(html):
-    return BeautifulSoup(html, 'html.parser')
+def html_to_soup(url):
```

Comment:
seria esse o nome né?
```suggestion
def url_to_soup(url):
```